src/main/java/kostka/moviecatalog/service/StatisticService.java

```diff
@@ -0,0 +1,119 @@
+        LOGGER.info("incrementing sync redis counter finished, new value = {}.", getSyncedRedisCounterValue());
+    }
+
+
+
+    @Async
+    public void incrementSyncedRabbitMqCounter() {
+        LOGGER.info("incrementing sync rabbitMQ counter started, old value = {}.", getSyncedRabbitMqCounterValue());
+        rabbitMqSyncedCounter.syncedIncrement();
+        incrementMapWithKey(RABBIT_MQ_SYNCED_COUNTER);
+        LOGGER.info("incrementing sync rabbitMQ counter finished, new value = {}.", getSyncedRabbitMqCounterValue());
+    }
+
+    @Async
+    public void incrementSyncedElasticCounter() {
+        LOGGER.info("incrementing sync elastic counter started, old value = {}.", getSyncedElasticCounterValue());
+        elasticSyncedCounter.syncedIncrement();
+        incrementMapWithKey(ELASTIC_SYNCED_COUNTER);
+        LOGGER.info("incrementing sync elastic counter finished, new value = {}.", getSyncedElasticCounterValue());
+    }
+
+    @Async
+    public void incrementSyncedScheduledCounter() {
+        LOGGER.info("incrementing sync scheduled counter started, old value = {}.", getSyncedScheduledCounterValue());
+        scheduledSyncedCounter.syncedIncrement();
+        incrementMapWithKey(SCHEDULED_SYNCED_COUNTER);
+        LOGGER.info("incrementing sync scheduled counter finished, new value = {}.", getSyncedScheduledCounterValue());
+    }
+
+    @Async
+    public void incrementSyncedDbCounter() {
+        LOGGER.info("incrementing sync scheduled counter started, old value = {}.", getSyncedDbCounterValue());
+        dbSyncedCounter.syncedIncrement();
+        incrementMapWithKey(DB_SYNCED_COUNTER);
+        LOGGER.info("incrementing sync scheduled counter finished, new value = {}.", getSyncedDbCounterValue());
+    }
+
+    public int getSyncedRedisCounterValue() {
+        return redisSyncedCounter.getCounterValue();
+    }
+
+    public int getSyncedScheduledCounterValue() {
+        return scheduledSyncedCounter.getCounterValue();
+    }
+
+    public int getSyncedElasticCounterValue() {
+        return elasticSyncedCounter.getCounterValue();
+    }
+
+    public int getSyncedRabbitMqCounterValue() {
+        return rabbitMqSyncedCounter.getCounterValue();
+    }
+
+    public int getSyncedDbCounterValue() {
+        return dbSyncedCounter.getCounterValue();
+    }
+
+    public StatisticDetail getAllStatistics() {
+        StatisticDetail detail = new StatisticDetail();
+        detail.setRedisCounter(this.getSyncedRedisCounterValue());
+        detail.setRabbitCounter(this.getSyncedRabbitMqCounterValue());
+        detail.setElasticCounter(this.getSyncedElasticCounterValue());
+        detail.setScheduledCounter(this.getSyncedScheduledCounterValue());
+        detail.setDbCounter(this.getSyncedDbCounterValue());
+        return detail;
+    }
+
+    public Map<String, Integer> getStatisticMap() {
+        return concurrentMap;
+    }
+
+    @Async
+    public void incrementMapWithKey(final String key) {
```

Comment:
Tahle metoda podle mě nebude thread-safe :)
 Protože máš tu inkrementaci rozdělenou do několika kroků, které se v několika vláknech můžou zase proložit špatným způsobem.

Totiž, i když jednotlivé operace na ConcurrentHashMap jsou v pohodě (get, remove), tak takováto kombinace už ne.

Reply:
díky :) upravím